src/bp/ui-studio/src/web/views/OneFlow/SidePanel/GoalEditor/export.ts

```diff
@@ -0,0 +1,82 @@
+import axios from 'axios'
+import 'bluebird-global'
+import sdk from 'botpress/sdk'
+import _ from 'lodash'
+
+const getActions = async (actionNames: string[]) => {
+  try {
+    const getAction = async (actionName: string) => {
+      const isGlobal = actionName.indexOf('/') > 0
+
+      const { data } = await axios.post(`${window.BOT_API_PATH}/mod/code-editor/readFile`, {
+        botId: !isGlobal && window.BOT_ID,
+        location: `${actionName}.js`,
+        type: 'action'
+      })
+
+      return data?.fileContent
+    }
+
+    return Promise.mapSeries(actionNames, async actionName => ({
+      actionName,
+      fileContent: await getAction(actionName)
+    }))
+  } catch (err) {
+    console.error(`Can't export actions: ${err}`)
+    return []
+  }
+}
+
+const getIntents = async (intentNames: string[]) => {
+  try {
+    const getIntent = async (intentName: string) => {
+      const { data } = await axios.get(`${window.BOT_API_PATH}/mod/nlu/intents/${intentName}`)
+      return data
+    }
+
+    return Promise.mapSeries(intentNames, async intent => getIntent(intent))
+  } catch (err) {
+    console.error(`Can't export intents: ${err}`)
+    return []
+  }
+}
+
+const getContentElements = async (ids: string[]): Promise<Partial<sdk.ContentElement[]>> => {
```

Comment:
Should  you use `Pick<... , 'id' | 'contentType', 'formData', 'previews'>` instead of `Partial` here?

Reply:
Yup you're right, fixed !